packages/catalog-readmodel-writer/src/model/converterV2.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  AgreementApprovalPolicy,
+  AgreementApprovalPolicyV2,
+  EServiceAttribute,
+  Descriptor,
+  DescriptorState,
+  Document,
+  EService,
+  EServiceAttributeV2,
+  EServiceDescriptorStateV2,
+  EServiceDescriptorV2,
+  EServiceDocumentV2,
+  EServiceTechnologyV2,
+  EServiceV2,
+  Technology,
+  agreementApprovalPolicy,
+  descriptorState,
+  technology,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { P, match } from "ts-pattern";
+
+export const fromAgreementApprovalPolicyV2 = (
+  input: AgreementApprovalPolicyV2 | undefined
+): AgreementApprovalPolicy | undefined => {
+  if (input == null) {
+    return undefined;
+  }
+
+  switch (input) {
+    case AgreementApprovalPolicyV2.MANUAL:
+      return agreementApprovalPolicy.manual;
+    case AgreementApprovalPolicyV2.AUTOMATIC:
+      return agreementApprovalPolicy.automatic;
+  }
+};
+
+export const fromEServiceDescriptorStateV2 = (
+  input: EServiceDescriptorStateV2
+): DescriptorState => {
+  switch (input) {
+    case EServiceDescriptorStateV2.DRAFT:
+      return descriptorState.draft;
+    case EServiceDescriptorStateV2.SUSPENDED:
+      return descriptorState.suspended;
+    case EServiceDescriptorStateV2.ARCHIVED:
+      return descriptorState.archived;
+    case EServiceDescriptorStateV2.PUBLISHED:
+      return descriptorState.published;
+    case EServiceDescriptorStateV2.DEPRECATED:
+      return descriptorState.deprecated;
+  }
+};
+
+export const fromEServiceTechnologyV2 = (
+  input: EServiceTechnologyV2
+): Technology => {
+  switch (input) {
+    case EServiceTechnologyV2.REST:
+      return technology.rest;
+    case EServiceTechnologyV2.SOAP:
+      return technology.soap;
+  }
+};
+
+export const fromEServiceAttributeV2 = (
+  input: EServiceAttributeV2
+): EServiceAttribute[] =>
+  match<EServiceAttributeV2, EServiceAttribute[]>(input)
+    .with(
+      {
+        single: P.not(P.nullish),
+      },
+      ({ single }) => [{ ...single, id: unsafeBrandId(single.id) }]
+    )
+    .otherwise(() =>
+      input.group.map((a) => ({ ...a, id: unsafeBrandId(a.id) }))
+    );
+
+export const fromDocumentV2 = (input: EServiceDocumentV2): Document => ({
+  ...input,
+  id: unsafeBrandId(input.id),
+  uploadDate: new Date(input.uploadDate),
+});
+
+export const fromDescriptorV2 = (input: EServiceDescriptorV2): Descriptor => ({
+  ...input,
+  id: unsafeBrandId(input.id),
+  version: input.version.toString(),
+  attributes:
+    input.attributes != null
+      ? {
+          certified: input.attributes.certified.map(fromEServiceAttributeV2),
+          declared: input.attributes.declared.map(fromEServiceAttributeV2),
+          verified: input.attributes.verified.map(fromEServiceAttributeV2),
+        }
+      : {
+          certified: [],
+          declared: [],
+          verified: [],
+        },
```

Comment:
being mandatory, can this be changed to
```suggestion
       {
          certified: input.attributes.certified.map(fromEServiceAttributeV2),
          declared: input.attributes.declared.map(fromEServiceAttributeV2),
          verified: input.attributes.verified.map(fromEServiceAttributeV2),
        },
```
?

Reply:
No, because every message field is translated as a `T | undefined` type in typescript